src/settings.c

```diff
@@ -35,8 +35,11 @@
 static GHashTable *settings;
 static FcMonitor *fontconfig_monitor;
 static int fontconfig_serial;
+static ShellIntrospect *shell_introspect;
```

Comment:
There is already a singleton in `shellintrospect.c`, so instead of replicating that here, just use `shell_introspect_get()` where you don't already have it (i.e. in `on_settings_changed()`)